frost-secp256k1-tr/src/lib.rs

```diff
@@ -758,12 +758,10 @@ pub mod keys {
         fn tweak<T: AsRef<[u8]>>(self, merkle_root: Option<T>) -> Self {
             let t = tweak(&self.verifying_key().to_element(), merkle_root);
             let tp = ProjectivePoint::GENERATOR * t;
-            let public_key_package = self.into_even_y(None);
-            let verifying_key =
-                VerifyingKey::new(public_key_package.verifying_key().to_element() + tp);
+            let verifying_key = VerifyingKey::new(self.verifying_key().to_element() + tp);
```

Comment:
I thought this was needed? BIP-341 uses

```
P = lift_x(int_from_bytes(pubkey))
Q = point_add(P, point_mul(G, t))
```

where `lift_x` always returns an even-Y point. Am I missing something?

Reply:
I think you're right, this change might have been a mistake. It wouldn't have shown in the signing tests, because the FROST signatures are still valid under the output pubkey, but it would mean the taproot spending path would be unusable, because the "internal key" (`pubkey` in your cited example) doesn't result in the same point X-coordinate as the one put on-chain.

I'll add a test to confirm this is this the case, and then fix it if so.

Reply:
Fixed in 5dc16a29df7849b93e7184b9fb09c56e9177406a, thanks for spotting this :)

Reply:
Fixed in 5dc16a29df7849b93e7184b9fb09c56e9177406a - Thanks for spotting this :)